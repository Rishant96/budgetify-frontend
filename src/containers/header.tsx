import React, { useContext } from "react";
import { Header } from "../components";
import * as ROUTES from "../constants/routes";
import { FirebaseContext } from "../context/firebase";

type HeaderContainerProps = {
  isUser: boolean;
  isLogin?: boolean;
};

export const HeaderContainer: React.FC<HeaderContainerProps> = ({
  children,
  isUser,
  isLogin = false,
  ...restProps
}) => {
  const { firebase } = useContext(FirebaseContext);

  const link = !isUser ? (
    isLogin ? (
      <Header.Link to={ROUTES.LANDING}>Sign Up</Header.Link>
    ) : (
      <Header.Link to={ROUTES.SIGN_IN}>Login</Header.Link>
    )
  ) : (
    <>
      <Header.Link to={ROUTES.HOME}>Home</Header.Link>
      <Header.TextLink
        onClick={() => {
          firebase!.auth().signOut();
        }}
      >
        Logout
      </Header.TextLink>
    </>
  );

  return (
    <>
      <Header>
        <Header.Brand>
          <Header.BrandLink>Budgetify</Header.BrandLink>
        </Header.Brand>
        <Header.LinkBelt>{link}</Header.LinkBelt>
      </Header>
      {children}
    </>
  );
};
