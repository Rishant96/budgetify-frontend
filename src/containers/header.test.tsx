import React from "react";
import { shallow } from "enzyme";
import { HeaderContainer } from "./header";

describe("HeaderContainer", () => {
  it("should render my component", () => {
    const wrapper = shallow(<HeaderContainer isUser={false} />);
  });

  it("should render initial layout", () => {
    const component = shallow(<HeaderContainer isUser={false} />);
    expect(component.getElements()).toMatchSnapshot();
  });
});
