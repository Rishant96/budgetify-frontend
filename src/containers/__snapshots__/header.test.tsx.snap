exports[`HeaderContainer should render initial layout 1`] = `
Array [
<div class="headerstyles__Navbar-sc-1u2hqr-0 gGToPh">
<h1 class="headerstyles__Brand-sc-1u2hqr-1 lbbjsy">
<a class="headerstyles__BrandLink-sc-1u2hqr-2 jacLXc" href="/">Budgetify</a>
</h1>
<div class="headerstyles__LinkBelt-sc-1u2hqr-3 djZFQx">
<a class="headerstyles__Link-sc-1u2hqr-4 cBmRXL" href="/signin">Login</a>
</div>
</div>,
]
`;